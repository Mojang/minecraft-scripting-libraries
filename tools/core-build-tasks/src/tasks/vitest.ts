import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
