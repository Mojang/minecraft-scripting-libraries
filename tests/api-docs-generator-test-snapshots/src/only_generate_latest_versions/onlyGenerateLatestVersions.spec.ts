// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Only Generate Latest Script Module Versions', () => {
    it('Generates correct output containing only the latest module for each major release', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['ts', 'msdocs', 'changelog', 'changelog-json'],
            additionalArgs: `--include-modules latest`,
        });
    });
});
