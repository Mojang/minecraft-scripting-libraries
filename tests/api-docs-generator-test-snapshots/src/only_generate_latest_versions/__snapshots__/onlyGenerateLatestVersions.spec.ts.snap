// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Only Generate Latest Script Module Versions > Generates correct output containing only the latest module for each major release > # Of Files Generated 1`] = `10`;

exports[`Only Generate Latest Script Module Versions > Generates correct output containing only the latest module for each major release > changelog/test-module.json 1`] = `
"[
  {
    "version": "2.1.0-beta",
    "dependencies": [],
    "peer_dependencies": [],
    "classes": [],
    "interfaces": [],
    "errors": [],
    "objects": [],
    "functions": [],
    "constants": [
      {
        "name": "GlobalConstantV210Beta",
        "$added": true,
        "is_read_only": true,
        "is_static": true,
        "type": {
          "is_bind_type": false,
          "is_errorable": false,
          "name": "int32"
        },
        "value": 8
      }
    ],
    "enums": [],
    "type_aliases": []
  },
  {
    "version": "2.0.0",
    "dependencies": [],
    "peer_dependencies": [],
    "classes": [],
    "interfaces": [],
    "errors": [],
    "objects": [],
    "functions": [],
    "constants": [
      {
        "name": "GlobalConstantV200",
        "$added": true,
        "is_read_only": true,
        "is_static": true,
        "type": {
          "is_bind_type": false,
          "is_errorable": false,
          "name": "int32"
        },
        "value": 8
      }
    ],
    "enums": [],
    "type_aliases": []
  },
  {
    "version": "1.1.0",
    "dependencies": [],
    "peer_dependencies": [],
    "classes": [],
    "interfaces": [],
    "errors": [],
    "objects": [],
    "functions": [],
    "constants": [
      {
        "name": "GlobalConstantV110",
        "$added": true,
        "is_read_only": true,
        "is_static": true,
        "type": {
          "is_bind_type": false,
          "is_errorable": false,
          "name": "int32"
        },
        "value": 8
      }
    ],
    "enums": [],
    "type_aliases": []
  },
  {
    "version": "1.0.0",
    "$added": true,
    "constants": [
      {
        "is_read_only": true,
        "is_static": true,
        "name": "GlobalConstantV100",
        "type": {
          "is_bind_type": false,
          "is_errorable": false,
          "name": "int32"
        },
        "value": 8
      }
    ],
    "minecraft_version": "1.0.0",
    "name": "test-module",
    "uuid": "a05a18e6-279a-4d87-862d-2a85442a2aed",
    "module_type": "script"
  }
]"
`;

exports[`Only Generate Latest Script Module Versions > Generates correct output containing only the latest module for each major release > msdocs/priorscriptapi/TOC.yml 1`] = `
"  - name: test-module 1.x.x
    items:
    - name: test-module 1.x.x
      href: test-module-1xx/test-module.md
"
`;

exports[`Only Generate Latest Script Module Versions > Generates correct output containing only the latest module for each major release > msdocs/priorscriptapi/test-module-1xx/test-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: test-module-1xx Module
description: Contents of the test-module module (Version 1.x.x)
---
# \`test-module\` Module (Version 1.x.x)

This should only be applied in the latest module!

> [!IMPORTANT]
> This documentation is for an older version of this module. Go to the latest documentation [*here*](../../../scriptapi/test-module/test-module.md).

## Manifest Details
\`\`\`json
{
    "module_name": "test-module",
    "version": "1.1.0"
}
\`\`\`

## Available Versions
- \`1.1.0\`

## Constants

### **GlobalConstantV100**
\`static read-only GlobalConstantV100 = 8;\`

This should only be applied in the latest module!

Type: *number*

### **GlobalConstantV110**
\`static read-only GlobalConstantV110 = 8;\`

This should only be applied in the latest module!

Type: *number*
"
`;

exports[`Only Generate Latest Script Module Versions > Generates correct output containing only the latest module for each major release > msdocs/scriptapi/TOC.yml 1`] = `
"  - name: test-module
    items:
    - name: test-module
      href: test-module/test-module.md
    - name: test-module Changelog
      href: test-module/changelog.md
"
`;

exports[`Only Generate Latest Script Module Versions > Generates correct output containing only the latest module for each major release > msdocs/scriptapi/test-module/changelog.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: test-module Changelog
description: Changelog of the \`test-module\` module
---
# \`test-module\` Changelog

## Version Changes
- [2.1.0-beta](#210-beta)
- [2.0.0](#200)
- [1.1.0](#110)
- [1.0.0](#100)

## 2.1.0-beta
#### Added constant \`GlobalConstantV210Beta\`
## 2.0.0
#### Added constant \`GlobalConstantV200\`
## 1.1.0
#### Added constant \`GlobalConstantV110\`
## 1.0.0
#### Added \`test-module\` Module
"
`;

exports[`Only Generate Latest Script Module Versions > Generates correct output containing only the latest module for each major release > msdocs/scriptapi/test-module/test-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: test-module Module
description: Contents of the test-module module
---
# \`test-module\` Module

This should only be applied in the latest module!

## [Changelog](changelog.md)

## Manifest Details
\`\`\`json
{
    "module_name": "test-module",
    "version": "2.0.0"
}
\`\`\`
This is version 2.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`2.1.0-beta\`
- \`2.0.0\`

### Prior Versions

The following API versions are from a major version that is documented here: [*test-module*](../../../priorscriptapi/test-module-1xx/test-module.md):
- \`1.1.0\`


## Constants

### **GlobalConstantV100**
\`static read-only GlobalConstantV100 = 8;\`

This should only be applied in the latest module!

Type: *number*

### **GlobalConstantV110**
\`static read-only GlobalConstantV110 = 8;\`

This should only be applied in the latest module!

Type: *number*

### **GlobalConstantV200**
\`static read-only GlobalConstantV200 = 8;\`

This should only be applied in the latest module!

Type: *number*

::: moniker range="=minecraft-bedrock-experimental"
### **GlobalConstantV210Beta**
\`static read-only GlobalConstantV210Beta = 8;\`

This should only be applied in the latest module!

Type: *number*

> [!CAUTION]
> This property is still in pre-release.  Its signature may change or it may be removed in future releases.
::: moniker-end
"
`;

exports[`Only Generate Latest Script Module Versions > Generates correct output containing only the latest module for each major release > txt/test-module.txt 1`] = `
"\`test-module\`

== Change Log ==
=== 2.1.0-beta ===
- Added constant \`GlobalConstantV210Beta\`

=== 2.0.0 ===
- Added constant \`GlobalConstantV200\`

=== 1.1.0 ===
- Added constant \`GlobalConstantV110\`

=== 1.0.0 ===
- Added \`test-module\` module

== /Change Log =="
`;

exports[`Only Generate Latest Script Module Versions > Generates correct output containing only the latest module for each major release > typescript/test-module@1.1.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 * This should only be applied in the latest module!
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "test-module",
 *   "version": "1.1.0"
 * }
 * \`\`\`
 *
 */
/**
 * @remarks
 * This should only be applied in the latest module!
 *
 */
export const GlobalConstantV100 = 8;
/**
 * @remarks
 * This should only be applied in the latest module!
 *
 */
export const GlobalConstantV110 = 8;
"
`;

exports[`Only Generate Latest Script Module Versions > Generates correct output containing only the latest module for each major release > typescript/test-module@2.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 * This should only be applied in the latest module!
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "test-module",
 *   "version": "2.0.0"
 * }
 * \`\`\`
 *
 */
/**
 * @remarks
 * This should only be applied in the latest module!
 *
 */
export const GlobalConstantV100 = 8;
/**
 * @remarks
 * This should only be applied in the latest module!
 *
 */
export const GlobalConstantV110 = 8;
/**
 * @remarks
 * This should only be applied in the latest module!
 *
 */
export const GlobalConstantV200 = 8;
"
`;

exports[`Only Generate Latest Script Module Versions > Generates correct output containing only the latest module for each major release > typescript/test-module@2.1.0-beta.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @beta
 * @packageDocumentation
 * This should only be applied in the latest module!
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "test-module",
 *   "version": "2.1.0-beta"
 * }
 * \`\`\`
 *
 */
/**
 * @remarks
 * This should only be applied in the latest module!
 *
 */
export const GlobalConstantV100 = 8;
/**
 * @remarks
 * This should only be applied in the latest module!
 *
 */
export const GlobalConstantV110 = 8;
/**
 * @remarks
 * This should only be applied in the latest module!
 *
 */
export const GlobalConstantV200 = 8;
/**
 * @beta
 * @remarks
 * This should only be applied in the latest module!
 *
 */
export const GlobalConstantV210Beta = 8;
"
`;
